### PR TITLE
list of professional connection requests

### PR DESCRIPTION
**Pull request by crcommons**
Converts ConnectionRequestPage to be used for visitor and organization member connection requests

### CODE REVIEW

**Comment by crcommons** [comment]
@cainlevy I was attempting to seed my local DB with everything needed to test this, but woah that's a lot. Wondering if there's a faster way for me to do this?

File: src/constants/Routes.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ type Domain =
   | "organizations"
   | "organizations/pending"
   | `organizations/${string}`
+  | "organizations/members/pending"
```
Comment (cainlevy) [inline src/constants/Routes.ts:45]: maybe `/connections/pending` or something similar?

**Review by cainlevy — COMMENTED** [review]
seems like a promising direction to me

File: src/pages/ConnectionRequestsPage.tsx [hunk 2]
```diff
@@ -41,13 +49,33 @@ const RequestsPage: React.FC = () => {
   const [connectionToReject, setConnectionToReject] =
     useState<ConnectionRequest>();
 
-  const connections = data?.facility.pendingConnectionRequests || [];
+  const connections =
+    data?.facility.pendingConnectionRequests.filter((requests) => {
+      if (
+        route.path === "/organizations/members/pending" &&
```
[the remark this answers is not in the document]
Comment (crcommons) [reply src/pages/ConnectionRequestsPage.tsx]: My issue with route params is that we don't have a usecase for all connection requests. If someone navigates to the page without the params, we don't know which connection type to query for or which content to display.

File: src/pages/ConnectionRequestsPage.tsx [hunk 3]
```diff
@@ -41,13 +49,33 @@ const RequestsPage: React.FC = () => {
   const [connectionToReject, setConnectionToReject] =
     useState<ConnectionRequest>();
 
-  const connections = data?.facility.pendingConnectionRequests || [];
+  const connections =
+    data?.facility.pendingConnectionRequests.filter((requests) => {
+      if (
+        route.path === "/organizations/connections/pending" &&
+        featureEnabled("PROVIDERS")
+      ) {
```
Comment (cainlevy) [inline src/pages/ConnectionRequestsPage.tsx:57]: this logic is repeated a few times. we could minimize our exposure to the path matching by doing it once?

**Comment by crcommons** [comment]
Apologies @hnqlv! I should have waited to merge my last PR into this until after you reviewed. You'll see some badge count code you're familiar with.

File: src/components/Requests/ConnectionRequestsTable.tsx [hunk 4]
```diff
@@ -57,40 +51,128 @@ const ConnectionRequestsTable = ({
...
+                      {membership.organization.name}
+                      <br></br>
+                    </span>
+                  </>
+                );
+              })
+            }
+          />
+        </>
+      ) : (
```
Comment (hnqlv) [inline src/components/Requests/ConnectionRequestsTable.tsx:81]: This is probably nitpicking but I'd rather have the following than a big block with a ternary:

```tsx
{ isProviders && (
   ...
)}

{ !isProviders && (
  ...
)}

```

**Review by hnqlv — APPROVED** [review]
@crcommons looks good! Just left two small notes 👍🏼

File: src/components/Requests/ConnectionRequestsTable.tsx [hunk 5]
```diff
@@ -57,40 +51,128 @@ const ConnectionRequestsTable = ({
...
+          <Table.Column
+            title="Organization"
+            dataIndex="visitor"
+            key="organizationMemberships"
+            render={(visitor: OrganizationMember) =>
+              // To Do: how do we want to handle displaying multiple organization memberships?
+              // To Do: Styling of org logo
+              visitor.organizationMemberships.map((membership: any) => {
+                return (
+                  <>
```
[the remark this answers is not in the document]
Comment (crcommons) [reply src/components/Requests/ConnectionRequestsTable.tsx:65]: Missed that one. But added a key

**Pull request merged**